shared/actions/favorite.js

```diff
@@ -172,23 +172,24 @@ export function favoriteList (): (dispatch: Dispatch) => void {
         dispatch(action)
         dispatch(badgeApp('newTLFs', !!(state.publicBadge || state.privateBadge)))
 
-        const newNotify = {public: state.publicBadge, private: state.privateBadge}
         const total = state.publicBadge + state.privateBadge
 
-        if (total && !_.isEqual(newNotify, previousNotify)) {
-          previousNotify = newNotify
+        if (total) {
+          const newNotifyState = [].concat(state.private.tlfs || [], state.public.tlfs || [])
+            .filter(t => t.meta === 'new').map(t => t.path)
 
-          let body
+          if (_.difference(newNotifyState, previousNotifyState).length) {
```

Comment:
shouldn't you be using the return value of difference to figure out the total and the body?